jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java

```diff
@@ -171,6 +168,118 @@ public void testCreateWithDefaultsAndExistingRouteWithMatchingNameInBuilderInOpe
+    }
+
+    @Test
+    public void testMergeRouteWithEmptyFragment() {
+        // Given
+        Route opinionatedRoute = getMockOpinionatedRoute();
+        Route fragmentRoute = new RouteBuilder().build();
+
+        // When
+        Route result = RouteEnricher.mergeRoute(fragmentRoute, opinionatedRoute);
+
+        // Then
+        assertNotNull(result);
+        assertEquals(opinionatedRoute, result);
+    }
+
+    @Test
+    public void testMergeRouteWithNonEmptyFragment() {
+        // Given
+        Route opinionatedRoute = getMockOpinionatedRoute();
+        Route fragmentRoute = new RouteBuilder()
+                .withNewSpec()
+                .withNewTls()
+                .withInsecureEdgeTerminationPolicy("Redirect")
+                .withTermination("edge")
+                .endTls()
+                .endSpec()
+                .build();
+
+        // When
+        Route result = RouteEnricher.mergeRoute(fragmentRoute, opinionatedRoute);
+
+        // Then
+        assertNotNull(result);
+        assertThat(result)
+                .extracting("metadata.name", "spec.host", "spec.to.kind", "spec.to.name",
+                        "spec.port.targetPort.intVal", "spec.tls.insecureEdgeTerminationPolicy", "spec.tls.termination")
+                .contains("test-svc", "example.com", "Service", "test-svc",
+                        8080, "Redirect", "edge");
+    }
+
+    private Route getMockOpinionatedRoute() {
+        return new RouteBuilder()
+                .withNewMetadata().withName("test-svc").endMetadata()
+                .withNewSpec()
+                .withNewPort()
+                .withNewTargetPort().withIntVal(8080).endTargetPort()
+                .endPort()
+                .withHost("example.com")
+                .withNewTo().withKind("Service").withName("test-svc").endTo()
+                .withNewTls()
+                .withInsecureEdgeTerminationPolicy("Redirect")
+                .withTermination("edge")
+                .endTls()
+                .addNewAlternateBackend()
+                .withKind("Service")
+                .withName("test-svc-2")
+                .withWeight(10)
+                .endAlternateBackend()
+                .endSpec()
+                .build();
+    }
+
+    private ServiceBuilder getMockServiceBuilder() {
+        return new ServiceBuilder()
+                .editOrNewMetadata()
+                .withName("test-svc")
+                .addToLabels("expose", "true")
+                .endMetadata()
+                .editOrNewSpec()
+                .addNewPort()
+                .withName("http")
+                .withPort(8080)
+                .withProtocol("TCP")
+                .withTargetPort(new IntOrString(8080))
+                .endPort()
+                .addToSelector("group", "test")
+                .withType("LoadBalancer")
+                .endSpec();
```

Comment:
If you preserve indentation. it's much easier to read.
Surround the code with `// @formatter:off` `// @formatter:on` comments so that autoformat won't accidentally break indentation.